### PR TITLE
[Application] Training Resnet kotlin code for Android @open sesame 03/20 14:42

### PR DESCRIPTION
**Pull request by jijoongmoon**
This PR includes the kotlin android implemetation to train Resnet18 in
Applications.

**Changes proposed in this PR:**
- Added TOC generator for README.md

Resolves:

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

File: Applications/Android/kotlin/app/src/main/java/com/samsung/sr/nntr/nntrainer/Trainer.kt [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+
+class Trainer(private val context: Context ) {
+
+    fun train() {
+    	val batch_size = 16
+	val epoch = 2
+
+	var model_bin_path = context.filesDir.toString()+"/weight.bin"
```
Comment (jihochu) [inline Applications/Android/kotlin/app/src/main/java/com/samsung/sr/nntr/nntrainer/Trainer.kt:23]: I'm not sure, but there are some kotlin style checkers.
https://github.com/pinterest/ktlint
and, vscode plugin: https://marketplace.visualstudio.com/items?itemName=esafirm.kotlin-formatter

Applying formatter would be better in the later.

File: Applications/Android/kotlin/app/src/main/jni/README.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+Inorder to build, you need nntrainer setup.
+please see https://github.com/nntstreamer/nntrainer/tree/main/docs
+
+Also you need to set NNTRAINER_ROOT parameter.
+
+``` bash
+export NNTRAINER_ROOT=$HOME/nntrainer
+```
+
+Also it is looking for the libs dir in NNTRAINER_ROOT. So you need copy the libs dir to $NNTRAINER_ROOT after android build.
```
Comment (jihochu) [inline Applications/Android/kotlin/app/src/main/jni/README.md:17]:
```suggestion
Also it is looking for the libs dir in NNTRAINER_ROOT. So you need to copy the libs dir to $NNTRAINER_ROOT after android build.
```

File: Applications/Android/kotlin/app/src/main/jni/cifar_dataloader.cpp [hunk 3]
```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2020 Jihoon Lee <jhoon.it.lee@samsung.com>
```
Comment (jihochu) [inline Applications/Android/kotlin/app/src/main/jni/cifar_dataloader.cpp:3]: If we use totally duplicated files, how about just linking files ?

File: Applications/Android/kotlin/app/src/main/java/com/samsung/sr/nntr/nntrainer/Trainer.kt (hunk 1 above)
Comment (jijoongmoon) [reply Applications/Android/kotlin/app/src/main/java/com/samsung/sr/nntr/nntrainer/Trainer.kt:23]: Great. I will check if there is available kotlin style checker.

File: Applications/Android/kotlin/app/src/main/jni/cifar_dataloader.cpp (hunk 3 above)
Comment (jijoongmoon) [reply Applications/Android/kotlin/app/src/main/jni/cifar_dataloader.cpp]: done. thanks.

File: Applications/Android/kotlin/app/src/main/jni/README.md (hunk 2 above)
Comment (jijoongmoon) [reply Applications/Android/kotlin/app/src/main/jni/README.md]: fixed. thanks.

File: Applications/Android/kotlin/app/src/main/java/com/samsung/sr/nntr/nntrainer/Trainer.kt [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+	var dummyArray : FloatArray = FloatArray(1)
+
+	var result = NNTrainerNative().nntrainerTest(dummyArray, outputArray, d_size, image_size, l_len, dir_path, model_bin_path, correct)
+
+
+	var ss =""
+	var right:Float = result.toFloat()
+	var accuracy = right /d_size *100.0
+
+	ss = "Accuracy : " +"%.3f".format(accuracy) +"% "+"%d".format(result)+"\n" + ss
```
Comment (baek2sm) [inline Applications/Android/kotlin/app/src/main/java/com/samsung/sr/nntr/nntrainer/Trainer.kt:57]: It is trivial but It seems good to add a space between the each variable and operator for code consistency(line 27-29, 31-32, 39-42, 44-46, 53-55, 57).

**Review by lhs8928 — APPROVED** [review]
LGTM

**Review by DonghakPark — APPROVED** [review]
LGTM!

**Review by baek2sm — APPROVED** [review]
LGTM except the above mentioned.

File: Applications/Resnet/PyTorch/main.py [hunk 5]
```diff
@@ -108,6 +108,17 @@ def forward(self, x):
         out = self.linear(out)
         return out
 
+    def save_bin(self, name):
```
[the remark this answers is not in the document]
Comment (jijoongmoon) [reply Applications/Resnet/PyTorch/main.py]: My bad. It does not relate to this PR. I remove this file.

File: Applications/Android/kotlin/app/src/main/java/com/samsung/sr/nntr/nntrainer/Trainer.kt (hunk 4 above)
Comment (jijoongmoon) [reply Applications/Android/kotlin/app/src/main/java/com/samsung/sr/nntr/nntrainer/Trainer.kt:49]: I'm sure there is a certain kotlin style checker and I will apply if I found the proper one.

**Pull request merged**